src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java

```diff
@@ -38,57 +39,52 @@ public class PasswordResetController {
             summary = "Смена пароля",
             description = "Смена пароля по ссылке"
     )
     @ApiResponses(value = {
-        @ApiResponse(responseCode = "400", description = "Введенный пароль некорректен",
+        @ApiResponse(responseCode = "409", description = "Пароли не совпадают",
             content = {@Content(mediaType = Constant.MEDIA_TYPE,
-                    schema = @Schema(implementation = ErrorListResponse.class))}),
+                    schema = @Schema(ref = "#/components/schemas/NewErrorBody"))}),
         @ApiResponse(responseCode = "403", description = "Ссылка недействительна",
             content = {@Content(mediaType = Constant.MEDIA_TYPE,
-                    schema = @Schema(implementation = ErrorListResponse.class))}),
-        @ApiResponse(responseCode = "200", description = "Пароль успешно изменен")
+                    schema = @Schema(ref = "#/components/schemas/NewErrorBody"))}),
+        @ApiResponse(responseCode = "201", description = "Пароль успешно изменен")
         }
     )
     @PostMapping("/update")
     public ResponseEntity<?> updatePassword(@RequestParam final String token,
                                             @Valid @RequestBody final UserPasswordDto userPasswordDto,
                                             final BindingResult bindingResult) {
-        final ErrorListResponse finalResponse = new ErrorListResponse();
-        if (!userPasswordDto.getPassword().equals(userPasswordDto.getPasswordConfirm())) {
-            finalResponse.getErrors().add("password: Пароли не совпадают");
-            return new ResponseEntity<>(finalResponse, HttpStatus.BAD_REQUEST);
-        }
         if (bindingResult.hasErrors()) {
-            bindingResult.getAllErrors().forEach(f -> finalResponse.getErrors()
-                    .add(Objects.requireNonNull(f.getDefaultMessage())));
-            return new ResponseEntity<>(finalResponse, HttpStatus.BAD_REQUEST);
-        }
-        final Optional<ErrorListResponse> response = resetPasswordService.updatePassword(token, userPasswordDto);
-        if (response.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.OK);
-        } else {
-            return new ResponseEntity<>(response.get(), HttpStatus.FORBIDDEN);
+            throw new PasswordsDontMatchException();
         }
+        resetPasswordService.updatePassword(token, userPasswordDto);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
+    }
+
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @ExceptionHandler(PasswordsDontMatchException.class)
+    public Map<String, String> passwordConflict(final PasswordsDontMatchException exc) {
+        return Collections.singletonMap("password", exc.getMessage());
+    }
+
+    @ResponseStatus(HttpStatus.FORBIDDEN)
+    @ExceptionHandler(TokenInvalidException.class)
+    public Map<String, String> tokenInvalid(final TokenInvalidException exc) {
+        return Collections.singletonMap("token", exc.getMessage());
```

Comment:
Consider to restrict usage of obsolete Collections methods.

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-checkstyle-plugin</artifactId>
                <dependencies>
                    <dependency>
                        <groupId>com.puppycrawl.tools</groupId>
                        <artifactId>checkstyle</artifactId>
                        <version>${plugins.checkstyle.puppycrawl.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>com.thomasjensen.checkstyle.addons</groupId>
                        <artifactId>checkstyle-addons</artifactId>
                        <version>${plugins.checkstyle.thomasjensen.version}</version>
                    </dependency>
                </dependencies>

                                <module name="IllegalMethodCall">
                                    <property name="illegalMethodNames" value="emptyList, emptyMap, emptySet, emptyNavigableMap, emptyNavigableSet,
                                    emptySortedMap, emptySortedSet, singleton, singletonList, singletonMap"/>
                                </module>

Reply:
Maven refuses to find it

Reply:
<dependency>
    <groupId>com.thomasjensen.checkstyle.addons</groupId>
    <artifactId>checkstyle-addons</artifactId>
    <version>7.0.1</version>
</dependency>

Reply:
Dependency 'com.thomasjensen.checkstyle.addons:checkstyle-addons:7.0.1' not found

Reply:
Just update all deps
```
                <dependencies>
                    <dependency>
                        <groupId>com.puppycrawl.tools</groupId>
                        <artifactId>checkstyle</artifactId>
                        <version>10.3.1</version>
                    </dependency>
                    <dependency>
                        <groupId>com.thomasjensen.checkstyle.addons</groupId>
                        <artifactId>checkstyle-addons</artifactId>
                        <version>7.0.1</version>
                    </dependency>
                </dependencies>
```

Reply:
Added